examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,149 @@
+!unzip -q data.zip
+"""
+
+train_source = sorted(glob("data/source/*.jpg"))[:-60]
+train_targets = sorted(glob("data/target/*.jpg"))[:-60]
+
+val_source = sorted(glob("data/source/*.jpg"))[-60:]
+val_targets = sorted(glob("data/target/*.jpg"))[-60:]
+
+height, width = 352, 608
+
+
+def resize_with_pad(image, h, w):
+    return tf.image.resize_with_pad(image=image, target_height=h, target_width=w)
+
+
+def data_preprocess(source, target):
+    source_image = tf.io.decode_png(tf.io.read_file(source), channels=3)
+    target_image = tf.io.decode_png(tf.io.read_file(target), channels=3)
+    source_image = resize_with_pad(source_image, h=height, w=width) / 255.0
+    target_image = resize_with_pad(target_image, h=height, w=width) / 255.0
+    return source_image, target_image
+
+
+batch_size = 4
+buffer_size = int(len(train_source) * 0.2)
+train_pipe = tf.data.Dataset.from_tensor_slices((train_source, train_targets))
+train_pipe = train_pipe.map(data_preprocess, tf.data.AUTOTUNE).cache()
+train_pipe = train_pipe.shuffle(buffer_size).batch(batch_size).repeat()
+
+val_pipe = tf.data.Dataset.from_tensor_slices((val_source, val_targets))
+val_pipe = val_pipe.map(data_preprocess, tf.data.AUTOTUNE).cache()
+val_pipe = val_pipe.shuffle(buffer_size).batch(batch_size).repeat()
+
+
+train_step = len(train_source) // batch_size
+val_step = len(val_source) // batch_size
+
+fig, ax = plt.subplots(1, 2, figsize=(16, 8))
+ax = ax.flatten()
+for i, f in enumerate(val_pipe.take(1)):
+    noise, clean = f
+    noise = noise[i].numpy() * 255
+    clean = clean[i].numpy() * 255
+    ax[0].imshow(noise.astype(np.int))
+    ax[0].set_xlabel("source")
+    ax[1].imshow(clean.astype(np.int))
+    ax[1].set_xlabel("target")
+fig.text(x=0.45, y=0.78, s="Dataset Visualization", fontweight="bold")
+fig.set_tight_layout(None)
+
+
+def conv_unit(x, filters=64, padding="same"):
+    x = layers.Conv2D(filters=filters, kernel_size=3, padding=padding)(x)
+    x = layers.BatchNormalization()(x)
+    x = layers.Activation("relu")(x)
+    return x
+
+
+def subpixel_upsample(x, target_channels, upscale_ratio):
+    estimated_filters = target_channels * (upscale_ratio ** 2)
+    x = layers.Conv2D(estimated_filters, 3, padding="same")(x)
+    x = layers.BatchNormalization()(x)
+    x = layers.LeakyReLU(0.2)(x)
+    x = tf.nn.depth_to_space(x, block_size=upscale_ratio)
+    return x
+
+
+def upscale_unit(previous_input, current_input, target_channels=3, upscale_ratio=2):
+    previous_low_scale = subpixel_upsample(
+        previous_input, target_channels=target_channels, upscale_ratio=upscale_ratio
+    )
+    current_high_scale = conv_unit(current_input, filters=target_channels)
+    concat_scales = tf.concat((previous_low_scale, current_high_scale), axis=-1)
+    feature_manifold = conv_unit(concat_scales, filters=target_channels)
+    return feature_manifold
+
+
+def denoiser():
+    base_model = ResNet50V2(include_top=False, input_shape=(352, 608, 3))
```

Comment:
Please add `weights="imagenet"` to make it explicit that this is a pretrained model